crates/bevy_render/src/color.rs

```diff
@@ -37,6 +37,19 @@ impl Color {
     pub const fn rgba(r: f32, g: f32, b: f32, a: f32) -> Color {
         Color { r, g, b, a }
     }
+
+    pub fn hex<T: AsRef<[u8]>>(hex: T) -> Color {
+        let mut buf = [0; 3];
+        if hex::decode_to_slice(hex, &mut buf).is_ok() {
```

Comment:
Can we also add support for alpha hex values here? We could just add a branch that runs the alpha code if the input is the correct length.

Reply:
Alpha usually uses `0.0 - 1.0` or `0 - 255`, If use `hex` to describe it, it may be a bit confusing.
To add alpha, I think `rgba`  function should be used

Just like CSS:
```css
color: #FFFFFF;
/* alpha */
color: rgba(255, 255, 255, 0.5);
```

What do you think? @cart

Reply:
Modern css supports hex with rgba. Ex: RRGGBBAA. Many image/svg editors also support that format.

https://css-tricks.com/8-digit-hex-codes/

Reply:
I think we should eventually support all the hex formats of CSS: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#Syntax_2: `#RRGGBB[AA]` or `#RGB[A]`.

I think it'd be nice to upgrade that slightly be allowing the `#` to be optional, e.g.: `[#]RRGGBB[AA]` or `[#]RGB[A]`.